packages/flutter/lib/src/rendering/shifted_box.dart

```diff
@@ -628,71 +635,70 @@ class RenderConstrainedOverflowBox extends RenderAligningShiftedBox {
   }
 }
 
-/// Renders a box, imposing no constraints on its child, allowing the child to
-/// render at its "natural" size.
+/// A [RenderBox] that applies an arbitrary transform to its [constraints]
+/// before sizing its child using the new constraints, treating any overflow as
+/// error.
 ///
-/// This allows a child to render at the size it would render if it were alone
-/// on an infinite canvas with no constraints. This box will then attempt to
-/// adopt the same size, within the limits of its own constraints. If it ends
-/// up with a different size, it will align the child based on [alignment].
-/// If the box cannot expand enough to accommodate the entire child, the
-/// child will be clipped.
+/// This [RenderBox] sizes its child using a [BoxConstraints] created by
+/// applying [constraintsTransform] to this [RenderBox]'s own [constraints].
+/// This box will then attempt to adopt the same size, within the limits of its
+/// own constraints. If it ends up with a different size, it will align the
+/// child based on [alignment]. If the box cannot expand enough to accommodate
+/// the entire child, the child will be clipped if [clipBehavior] is not
+/// [Clip.none].
 ///
 /// In debug mode, if the child overflows the box, a warning will be printed on
 /// the console, and black and yellow striped areas will appear where the
 /// overflow occurs.
 ///
+/// This [RenderBox] allows the child to selectively enforce some of its
```

Comment:
what do you mean by "selectively enforcing some of its intrinsic sizing rules"?

Reply:
reworded.